lib/utils/import/googlekeep/index.ts

```diff
@@ -0,0 +1,101 @@
+import { EventEmitter } from 'events';
+import { endsWith, isEmpty, get, has } from 'lodash';
+import JSZip from 'jszip';
+
+import CoreImporter from '../';
+
+let fs = null;
+const isElectron = has(window, 'process.type');
+if (isElectron) {
+  fs = __non_webpack_require__('fs'); // eslint-disable-line no-undef
+}
+
+class GoogleKeepImporter extends EventEmitter {
+  constructor({ noteBucket, tagBucket, options }) {
+    super();
+    this.noteBucket = noteBucket;
+    this.tagBucket = tagBucket;
+    this.options = options;
+  }
+
+  importNotes = filesArray => {
+    if (isEmpty(filesArray)) {
+      this.emit('status', 'error', 'No file to import.');
+      return;
+    }
+
+    const coreImporter = new CoreImporter({
+      noteBucket: this.noteBucket,
+      tagBucket: this.tagBucket,
+    });
+
+    let importedNoteCount = 0;
+
+    const importJsonString = jsonString => {
+      const importedNote = JSON.parse(jsonString);
+
+      const title = importedNote.title;
+      let textContent = title ? title + '\n\n' : '';
+      textContent += get(importedNote, 'textContent', '');
+
+      if (importedNote.listContent) {
+        // Note has checkboxes
+        textContent += importedNote.listContent
+          .map(item => `- [${item.isChecked ? 'x' : ' '}] ${item.text}`)
+          .join('\n');
+      }
+
+      return coreImporter
+        .importNote(
+          {
+            content: textContent,
+            // note: the exported files don't tell us the creation date...
+            modificationDate: importedNote.userEditedTimestampUsec / 1e6,
+            pinned: importedNote.isPinned,
+            tags: get(importedNote, 'labels', []).map(item => item.name),
+          },
+          { ...this.options, isTrashed: importedNote.isTrashed }
```

Comment:
if it were me I might suggest sending archived notes to the trash _or_ add a tag to them - `archived` and leave them in the inbox.

trash is generally safe but someone could accidentally "empty the trash" and wipe out their archive

if, on the other hand, they import their archive into the "All Notes" section then they might get more notes in the list than they expect.

maybe we just need a setting…

```
Archived Notes: [ ] Import into trash [ ] Import with tag _________
```

Reply:
Yes, this is how I would do it as well. I don't know how to add a (importer-specific) setting to the dialog.

Reply:
I will try to play around with it this week and see if it wouldn't be too hard. if not, maybe adding `archive` to the tags could be a good interim solution?